src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs

```diff
@@ -315,7 +317,9 @@ private bool IsInVarContext(NameSyntax name)
             var identifierName = name as IdentifierNameSyntax;
             if (symbolInfo.Symbol.IsImplicitValueParameter())
             {
+#nullable disable
```

Comment:
Why?

Reply:
I've found this is a really clean solution for cases where I don't know how we want to handle it.

Reply:
The reported warning says `identifierName` may be null here. I can't tell if this is an invariant (in which case replace `as` with a cast) or something that needs new validation.

Reply:
Yes, if the symbol bound was the implicit value parameter, then it means yes it had to be an identifier name since you had to bind precisely "name". Adding an assert or Contract.ThrowIfNull() seems most appropriate here.

Reply:
Updated to exclude this block for now.